code/game/objects/structures/reflector.dm

```diff
@@ -56,83 +62,78 @@
+				if (do_after(user,20*W.toolspeed, target = src))
+					if(!src || !WT.isOn())
+						return
+					anchored = TRUE
+					to_chat(user, "<span class='notice'>You weld \the [src] to the floor.</span>")
+		else
+			if (WT.remove_fuel(0,user))
+				playsound(src.loc, 'sound/items/welder2.ogg', 50, 1)
+				user.visible_message("[user.name] starts to cut the [src.name] free from the floor.", \
+					"<span class='notice'>You start to cut \the [src] free from the floor...</span>", \
+					"<span class='italics'>You hear welding.</span>")
+				if (do_after(user,20*W.toolspeed, target = src))
+					if(!src || !WT.isOn())
+						return
+					anchored = FALSE
+					to_chat(user, "<span class='notice'>You cut \the [src] free from the floor.</span>")
 	//Finishing the frame
 	else if(istype(W, /obj/item/stack/sheet))
 		if(finished)
 			return
 		var/obj/item/stack/sheet/S = W
 		if(istype(W, /obj/item/stack/sheet/glass))
-			if(S.get_amount() < 5)
-				to_chat(user, "<span class='warning'>You need five sheets of glass to create a reflector!</span>")
-				return
-			else
-				S.use(5)
+			if(S.use(5))
 				new /obj/structure/reflector/single (src.loc)
 				qdel (src)
-		if(istype(W, /obj/item/stack/sheet/rglass))
-			if(S.get_amount() < 10)
-				to_chat(user, "<span class='warning'>You need ten sheets of reinforced glass to create a double reflector!</span>")
-				return
 			else
-				S.use(10)
+				to_chat(user, "<span class='warning'>You need five sheets of glass to create a reflector!</span>")
+				return
+		if(istype(W, /obj/item/stack/sheet/rglass))
+			if(S.use(10))
 				new /obj/structure/reflector/double (src.loc)
 				qdel(src)
+			else
+				to_chat(user, "<span class='warning'>You need ten sheets of reinforced glass to create a double reflector!</span>")
+				return
 		if(istype(W, /obj/item/stack/sheet/mineral/diamond))
-			if(S.get_amount() >= 1)
-				S.use(1)
+			if(S.use(1))
 				new /obj/structure/reflector/box (src.loc)
 				qdel(src)
 	else
 		return ..()
 
-/obj/structure/reflector/proc/get_reflection(srcdir,pdir)
-	return 0
-
-
-/obj/structure/reflector/verb/rotate()
-	set name = "Rotate"
-	set category = "Object"
-	set src in oview(1)
-
-	if(usr.stat || !usr.canmove || usr.restrained())
+/obj/structure/reflector/proc/rotate(mob/user)
+	if (anchored)
+		to_chat(user, "<span class='warning'>It is fastened to the floor!</span>")
+		return FALSE
+	var/new_angle = input(user, "Input a new angle for primary reflection face.", "Reflector Angle") as null|num
+	if(!user.canUseTopic(src, be_close=TRUE))
+		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
 		return
-	if (src.anchored)
-		to_chat(usr, "<span class='warning'>It is fastened to the floor!</span>")
-		return 0
-	src.setDir(turn(src.dir, 270))
-	return 1
+	setAngle(NORM_ROT(new_angle))
+	return TRUE
 
+/obj/structure/reflector/proc/setAngle(new_angle)
+	rotation_angle = new_angle
+	setDir(NORTH)
+	var/matrix/M = new
```

Comment:
should this not be `= matrix(transform)`?

Reply:
it resets transform to 0 degrees with this before rotating again.